python/ray/data/datasource/tensorflow_datasource.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from typing import TYPE_CHECKING, Callable, Iterator, List
+
+from ray.data.block import Block, BlockMetadata, T
+from ray.data.datasource import Datasource, ReadTask
+
+if TYPE_CHECKING:
+    import tensorflow as tf
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleTensorFlowDatasource(Datasource[T]):
+    """A datasource that let's you use TensorFlow datasets with Ray Data.
+
+    .. warning::
+        ``SimpleTensorFlowDataset`` doesn't support parallel reads. You should only use
+        this datasource for small datasets like MNIST or CIFAR.
+
+    Example:
+        >>> def dataset_factory():
+        ...     return tfds.load("cifar10", split=["train"], as_supervised=True)[0]
+        ...
```

Comment:
Not if this line is needed, but this is what it looks like if I use my Python interpreter.